src/backend/graphItemService.ts

```diff
@@ -148,154 +170,89 @@ export class GraphItemService {
       try {
         if (isEdge) {
           // 3.a. Build and run a query to create edges
-          const query = GraphItemService.buildEdgesQuery(params.itemType, headers, batch.rows);
+          const query = GraphItemService.buildEdgesQuery(params.itemType, propertyKeys, batch.rows);
           await GraphItemService.runCypherQuery(rc, query, params.sourceKey);
         } else {
           // 3.a. Build and run a query to create nodes
-          const query = GraphItemService.buildNodesQuery(params.itemType, headers, batch.rows);
+          const query = GraphItemService.buildNodesQuery(params.itemType, propertyKeys, batch.rows);
           const response = await GraphItemService.runCypherQuery(rc, query, params.sourceKey);
           GraphItemService.cacheGraphNodeIDs(params.itemType, response, batch.UIDs);
         }
       } catch (e) {
         log('Batch has failed', e);
         errors.add(e, batch.indices);
       }
-      this.updateProgress(i, total, errors);
+      this.updateImportProgress(totalRecords, i);
     }
 
     // LKE-4201 Remove the CSV_PLUGIN category
     await GraphItemService.runCypherQuery(rc,'MATCH(c:CSV_PLUGIN) DETACH DELETE c RETURN 0', params.sourceKey);
 
-    this.endProgress(total, errors);
+    this.finishImport(errors, totalRecords);
   }
 
-  /**
-   * Parse a CSV string value intro a string, number or boolean
-   */
-  public static parseValue(value: string): string | number | boolean | null {
-    // Properties without value
-    if (value === '') {
-      return null; // Cypher ignores null properties when creating
-    }
-    // Numerical properties
-    if (/^(0|([1-9]\d*))(\.\d+)?$/.test(value)) {
-      return Number.parseFloat(value);
-    }
-    // Boolean properties
-    if (value === 'true') {
-      return true;
-    }
-    if (value === 'false') {
-      return false;
-    }
-    return value;
-  }
-
-  /**
-   * TODO take care of escaping commas, line-breaks, etc...
-   * EOL is \n on POSIX and \r\n on Windows
-   */
-  public static createNodeBatches(csv: string): {
-    total: number;
-    headers: string[];
+  public static createNodeBatches(parsedCSV: ParsedCSV): {
+    propertyKeys: string[];
     batchedRows: {indices: number[]; rows: unknown[][]; UIDs: string[]}[];
-    badRows: [RowErrorMessage, number[]][];
   } {
     const MAX_BATCH_SIZE = 10;
-    let count = 1;
-    let headers: unknown[] | undefined = undefined;
     let batchedRows: {indices: number[]; UIDs: string[]; rows: unknown[][]}[] = [];
```

Comment:
why not using as a type Array<{records: Array<{index: number, UID: number, rows: unknown[]}>}>?

Reply:
I agree that other types like the one you suggest are easier to build and read, but this type `batchedRows: {indices: number[]; UIDs: string[]; rows: unknown[][]}[]` is easier to use, like in the following code:
```js
    for (const batch of batchedRows) {
      i += batch.rows.length;
      try {
        if (isEdge) {
          // 3.a. Build and run a query to create edges
          const query = GraphItemService.buildEdgesQuery(params.itemType, propertyKeys, batch.rows);
          await GraphItemService.runCypherQuery(rc, query, params.sourceKey);
        } else {
          // 3.a. Build and run a query to create nodes
          const query = GraphItemService.buildNodesQuery(params.itemType, propertyKeys, batch.rows);
          const response = await GraphItemService.runCypherQuery(rc, query, params.sourceKey);
          GraphItemService.cacheGraphNodeIDs(params.itemType, response, batch.UIDs);
        }
      } catch (e) {
        log('Batch has failed', e);
        errors.add(e, batch.indices);
      }
      this.updateImportProgress(totalRecords, i);
    }
 ```
having UIDs, rows and indices separated makes it easy to separate concerns in `errors.add()`, `cacheGraphNodeIDs()` and `buildNodesQuery()`